### PR TITLE
[android][tests] Tests for signed unsigned byte tensors

### PR DESCRIPTION
**Pull request by IvanKobzarev**
We expose int8 and uint8 tensors (Tensor.newUInt8Tensor, Tensor.newInt8Tensor). On java side we have only java.lang.Byte which is signed [-128, 127] 

Adding 2 tests:
torch.div(input, 2)
torch.add(input, 1)

unsigned value of byte is calculated as (int) (b & 0xFF)

### CODE REVIEW

File: android/pytorch_android/generate_test_torchscripts.py [hunk 1]
```diff
@@ -44,6 +44,16 @@ def eqTensor(self, input):
         # type: (Tensor) -> Tensor
         return input
 
+    @torch.jit.script_method
+    def tensorDiv2(self, input):
+        # type: (Tensor) -> Tensor
+        return torch.div(input, 2)
```
Comment (dreiss) [inline android/pytorch_android/generate_test_torchscripts.py]: Can this not be `return input / 2`?

File: android/pytorch_android/src/androidTest/java/org/pytorch/PytorchInstrumentedTests.java [hunk 2]
```diff
@@ -298,6 +299,105 @@ public void testStr3Concat() throws IOException {
...
+
+      final int expectedOutInt8I = i == Byte.MAX_VALUE ? Byte.MIN_VALUE : ((int) iB) + 1;
+      final byte expectedOutInt8B = (byte) expectedOutInt8I;
+      assertEqualsBytes(expectedOutInt8B, outInt8B);
+      assertEquals(expectedOutInt8I, (int) outInt8B);
+    }
+  }
+
+  private static int byteToUnsignedInt(byte b) {
+    return ((int) b) & 0xff;
```
Comment (dreiss) [inline android/pytorch_android/src/androidTest/java/org/pytorch/PytorchInstrumentedTests.java]: Should be consistent with `0xff` or `0xFF`.  (I prefer `0xFF`, but it's a matter of taste).

File: android/pytorch_android/src/androidTest/java/org/pytorch/PytorchInstrumentedTests.java [hunk 3]
```diff
@@ -298,6 +299,105 @@ public void testStr3Concat() throws IOException {
...
+
+  private static void assertEqualsBytes(byte expected, byte actual) {
+    if (expected != actual) {
+      assertTrue(String.format("Expected byte:%s(%s) actual byte:%s(%s)", Byte.toString(expected),
+          byteToBinaryString(expected), Byte.toString(actual), byteToBinaryString(actual)), false);
+    }
+  }
+
+  private static String byteToBinaryString(byte b) {
+    return String.format("%8s", Integer.toBinaryString(b & 0xFF)).replace(' ', '0');
```
Comment (dreiss) [inline android/pytorch_android/src/androidTest/java/org/pytorch/PytorchInstrumentedTests.java]: Does this work?

`return String.format("%08s", Integer.toBinaryString(b & 0xFF));`

File: android/pytorch_android/src/androidTest/java/org/pytorch/PytorchInstrumentedTests.java (hunk 2 above)
Comment (dreiss) [reply android/pytorch_android/src/androidTest/java/org/pytorch/PytorchInstrumentedTests.java]: Also, I don't think the cast is necessary.  All byte and char are converted to int before math operations.

File: android/pytorch_android/src/androidTest/java/org/pytorch/PytorchInstrumentedTests.java [hunk 4]
```diff
@@ -298,6 +299,105 @@ public void testStr3Concat() throws IOException {
...
+    final long[] inputTensorShape = new long[]{256};
+    final long numElements = Tensor.numel(inputTensorShape);
+    final byte[] inputTensorData = new byte[(int) numElements];
+    for (int i = 0; i < numElements; ++i) {
+      inputTensorData[i] = (byte) i;
+    }
+    final Module module = Module.load(assetFilePath(TEST_MODULE_ASSET_NAME));
+
+    final Tensor inputTensorUint8 = Tensor.newUInt8Tensor(inputTensorShape, inputTensorData);
+    final Tensor inputTensorInt8 = Tensor.newInt8Tensor(inputTensorShape, inputTensorData);
```
Comment (dreiss) [inline android/pytorch_android/src/androidTest/java/org/pytorch/PytorchInstrumentedTests.java]: If you change this to

```
    final Tensor inputTensorUint8 = Tensor.newInt8Tensor(inputTensorShape, inputTensorData);
    final Tensor inputTensorInt8 = Tensor.newUint8Tensor(inputTensorShape, inputTensorData);
```

Does the test fail?  If so, how?

**Review by dreiss — CHANGES_REQUESTED** [review]
Kicking back for questions.

File: android/pytorch_android/generate_test_torchscripts.py (hunk 1 above)
Comment (IvanKobzarev) [reply android/pytorch_android/generate_test_torchscripts.py]: Checked. It works the same.

File: android/pytorch_android/src/androidTest/java/org/pytorch/PytorchInstrumentedTests.java (hunk 3 above)
Comment (IvanKobzarev) [reply android/pytorch_android/src/androidTest/java/org/pytorch/PytorchInstrumentedTests.java]: Fails :)
```
java.util.FormatFlagsConversionMismatchException: Conversion = s, Flags = 0
at java.util.Formatter$FormatSpecifier.failMismatch(Formatter.java:4131)
at java.util.Formatter$FormatSpecifier.checkBadFlags(Formatter.java:3035)
at java.util.Formatter$FormatSpecifier.checkGeneral(Formatter.java:2993)
at java.util.Formatter$FormatSpecifier.<init>(Formatter.java:2764)
at java.util.Formatter$FormatSpecifierParser.<init>(Formatter.java:2591)
at java.util.Formatter.parse(Formatter.java:2524)
at java.util.Formatter.format(Formatter.java:2472)
at java.util.Formatter.format(Formatter.java:2426)
at java.lang.String.format(String.java:2670)
at org.pytorch.PytorchInstrumentedTests.byteToBinaryString(PytorchInstrumentedTests.java:358)
```

Also tried `"%0$8s"`, does not fail, but does not fill with 0s `Expected byte:2(      10) actual byte:1(       1)`

File: android/pytorch_android/src/androidTest/java/org/pytorch/PytorchInstrumentedTests.java (hunk 4 above)
Comment (IvanKobzarev) [reply android/pytorch_android/src/androidTest/java/org/pytorch/PytorchInstrumentedTests.java]: It passes :) No difference on binary representation. 
I will think about replacement for this test to have the difference.

**Comment by IvanKobzarev** [comment]
force pushed with update, sorry for that, starting using ghstack :)

 - tensorAdd2 replaced with tensorModulo3 (as python % is modulo, but java % works as a remainder, so if it is negative - adding the divisor to it)
If to change uint8 <-> int8  - it fails

 - 0xFF 

 - adding message to assertEqualsBytes to have assert failing more readable

 - removed redundant (int) conversions for byte

 - python script methods changed to:
```
tensor + scalarInt
tensor / scalarInt
tensor * scalarInt
tensor % scalarInt
```

**Pull request closed without merging**